public/app/features/dashboard/services/TimeSrv.ts

```diff
@@ -65,6 +70,51 @@ export class TimeSrv {
     }
   }
 
+  checkTimeAtLoad() {
+    let maxTimeRangeInMillis = 0,
+      oldestFromInMillis = 0;
+    if (this.dashboard?.timepicker.maxTimeRange) {
+      maxTimeRangeInMillis = rangeUtil.intervalToMs(this.dashboard.timepicker.maxTimeRange);
+    }
+    if (this.dashboard?.timepicker.oldestFrom) {
+      oldestFromInMillis = rangeUtil.intervalToMs(this.dashboard.timepicker.oldestFrom);
+    }
+    let min = 0;
+    if (maxTimeRangeInMillis !== 0 && oldestFromInMillis !== 0) {
+      min = Math.min(maxTimeRangeInMillis, oldestFromInMillis);
+    } else if (maxTimeRangeInMillis !== 0) {
+      min = maxTimeRangeInMillis;
+    } else if (oldestFromInMillis !== 0) {
+      min = oldestFromInMillis;
+    }
+    let str;
+    if (min !== 0) {
+      str = rangeUtil.secondsToHms(min / 1000);
+    }
+    const timezone = this.dashboard ? this.dashboard.getTimezone() : undefined;
+    let from: any = this.time.from;
+    let to: any = this.time.to;
+    if (dateMath.isMathString(from)) {
+      from = dateMath.parse(from, false, timezone);
+    }
+    if (dateMath.isMathString(to)) {
+      to = dateMath.parse(to, false, timezone);
+    }
+
+    if (
+      min !== 0 &&
+      ((maxTimeRangeInMillis !== 0 && to.valueOf() - from.valueOf() > maxTimeRangeInMillis) ||
+        (oldestFromInMillis !== 0 && Date.now() - from.valueOf() > oldestFromInMillis))
+    ) {
+      this.time.from = 'now-' + str;
+      this.time.to = 'now';
```

Comment:
Feels like this checkTimeAtLoad function has largely duplicated logic compared to the code in the setTime function, is there a way to consolidate them?